packages/fast-components-styles-msft/src/utilities/color/offsets-algorithm.ts

```diff
@@ -0,0 +1,62 @@
+import {
+    checkDesignSystemResolver,
+    DesignSystem,
+    DesignSystemResolver,
+} from "../../design-system";
+import { SwatchFamily, SwatchFamilyResolver } from "./common";
+import { findClosestBackgroundIndex, getSwatch, isDarkMode, Palette } from "./palette";
+
+/**
+ * Function to derive colors from offset configuration.
+ */
+export function offsetsAlgorithm(
+    palette: Palette | DesignSystemResolver<Palette>,
+    restDelta: number | DesignSystemResolver<number>,
+    hoverDelta: number | DesignSystemResolver<number>,
+    activeDelta: number | DesignSystemResolver<number>,
+    focusDelta: number | DesignSystemResolver<number>
```

Comment:
Just a thought, and I know this is in the idea stage - but perhaps this algo can be defined by more of a config driven aspect? That may help those using it and reading it. It seems like the algorithm needs two things: 1) Palette, 2) interactive state config for the deltas
```
export interface InteractiveOffsetConfig {
    restDelta: number | DesignSystemResolver<number>;
    hoverDelta: number | DesignSystemResolver<number>;
    activeDelta: number | DesignSystemResolver<number>;
    focusDelta: number | DesignSystemResolver<number>;
}

export function offsetsAlgorithm(palette: Palette | DesignSystemResolver<Palette>, config: InteractiveOffsetConfig): SwatchFamilyResolver { };
```

Reply:
Is the benefit here cleaner syntax, reuse of config, or both? It _does_ seem like it would protect from potential future changes easier as properties could be added and defaulted or deprecated without changing the signature. I would probably put everything in though, including the palette. I could see an implementation of this that wants both an accent and neutral, or even success or error, etc.

Perhaps in the design system service model there would be a way for style modules like this to retrieve properties from the registry of sorts. For instance, a custom design system might add a fourth color, and through composition the style could find it. This feels like a fairly standard composition problem that possibly has a framework or structure already in place. I wonder if we could pull something in.

Reply:
Initially I see it for cleaner syntax - from a fn() standpoint, once you hit more than 3 inputs it's going to become difficult to manage. Considering the deltas are all "related" and inform the palette I think the syntax makes more sense than having them all at the same level.